test/distributed/test_c10d_common.py

```diff
@@ -307,6 +313,360 @@ def _prepare_multi_device_module(
 
         return model, ddp_model, input, target
 
+    def _get_store(self):
+        return dist.FileStore(self.file_name, self.world_size)
+
+    def _get_process_group(self):
+        raise ValueError("To be implemented by child class")
+
+    def _train_model(self, model, input_var, target, loss, run_checkpoint=False, use_reentrant=True):
+        model.train()
+        if run_checkpoint:
+            output = checkpoint(model, input_var, use_reentrant=use_reentrant)
+        else:
+            output = model(input_var)
+        l = loss(output, target)
+        l.backward()
+
+    def _test_ddp_checkpointing(
+        self,
+        input_model,
+        process_group,
+        use_bucket_view,
+        find_unused_parameters=False,
+        static_graph=False,
+        run_checkpoint=False,
+        use_reentrant=True,
+        allow_none_grads=False,
+    ):
+        # to reproduce the same training results
+        torch.cuda.set_device(self.rank)
```

Comment:
Do the effects of `set_device()` persist after the test and possibly affect other tests?
If so, should we consider using `with torch.cuda.device(self.rank):` (context manager)?

Reply:
It won't affect other tests because all tests are launched and run in their own subprocess, but in general it might be better to use the context manager.

Historically we've always used torch.cuda.set_device(self.rank) and it should be fine when the arg is self.rank because we usually do the GPU mapping that way.